activerecord/lib/active_record/associations.rb

```diff
@@ -1885,6 +1889,9 @@ def has_one(name, scope = nil, **options)
         #   belong_to  :note, query_constraints: [:organization_id, :note_id]
         def belongs_to(name, scope = nil, **options)
           reflection = Builder::BelongsTo.build(self, name, scope, options)
+          if reflection.counter_cache_column
```

Comment:
Should this go inside `BelongsTo`? That appears to be where the rest of counter cache things are done:

https://github.com/rails/rails/blob/a561dd4b2928459feb313b2db0910d2bc0133864/activerecord/lib/active_record/associations/builder/belongs_to.rb#L27

Reply:
Yeah I think this would be a much better place especially because there is 🤔 Wonder if we could reuse it
https://github.com/rails/rails/blob/a561dd4b2928459feb313b2db0910d2bc0133864/activerecord/lib/active_record/associations/builder/belongs_to.rb#L40

But first I'd like to explore an alternative solution which will build the list on `schema_cache` load to allow for all classes (parents & childs) to be fully loaded

Reply:
In the new PR I moved the logic to the `BelongsTo` builder, thank you!